File: src/main/java/com/spagnola/lcars/panels/LMainPanel.java
package com.spagnola.lcars.panels;

import javax.swing.JButton;
import javax.swing.JLabel;
import javax.swing.JPanel;
import java.awt.event.ActionListener;

public class LMainPanel extends JPanel {

    protected JButton logoutButton;

    public LMainPanel(ActionListener logoutAction) {
        JLabel label = new JLabel("Main Panel");
        this.add(label);

        logoutButton = new JButton("LOGOUT");
        logoutButton.addActionListener(logoutAction);
        this.add(logoutButton);


    }


}

File: src/main/kotlin/de/visualdigits/lcars/panels/LMainPanel.kt
package de.visualdigits.lcars.panels

import java.awt.event.ActionListener
import javax.swing.JButton
import javax.swing.JLabel
import javax.swing.JPanel

class LMainPanel(logoutAction: ActionListener?) : JPanel() {
    private var logoutButton: JButton

    init {
        val label: JLabel = JLabel("Main Panel")
        this.add(label)

        logoutButton = JButton("LOGOUT")
        logoutButton.addActionListener(logoutAction)
        this.add(logoutButton)
    }
}
